Source/WTF/wtf/Vector.h

```diff
@@ -565,10 +565,13 @@ class VectorBuffer : private VectorBufferBase<T, Malloc> {
     {
         ASSERT_WITH_SECURITY_IMPLICATION(buffer());
 
-        IGNORE_GCC_WARNINGS_BEGIN("invalid-offsetof")
+#if OS(LINUX)
```

Comment:
Why only for Linux and not also for e.g. FreeBSD? Surely it's OK to always ignore the warning on any OS?

Reply:
Didn't think of FreeBSD :)
I thought if we ignore this on every OS and for every compiler, why do we have that code in the first place?

I guess it's useful on Apple platforms?

Reply:
The static assert is still compiled on every platform. Only the compiler warning is ignored. And that warning should be suppressed everywhere.

The result of the assertion is undefined behavior, but since it's only a compile time assert, I think it's OK to keep the assert. We'd just have to remove or disable it on platforms where it fails to build.

Reply:
OK I can remove the OS ifdef then, thanks :)